finetrainers/trainer.py

```diff
@@ -664,10 +723,20 @@ def train(self) -> None:
                     loss = loss.mean(list(range(1, loss.ndim)))
                     # Average loss across batch dimension
                     loss = loss.mean()
+                    print(f"{self.transformer.dtype=}, {accelerator.mixed_precision=}")
                     accelerator.backward(loss)
 
-                    if accelerator.sync_gradients and accelerator.distributed_type != DistributedType.DEEPSPEED:
-                        grad_norm = accelerator.clip_grad_norm_(self.transformer.parameters(), self.args.max_grad_norm)
+                    if accelerator.sync_gradients:
+                        if accelerator.distributed_type == DistributedType.DEEPSPEED:
+                            grad_norm = self.transformer.get_global_grad_norm()
+                            # In some cases the grad norm may not return a float
+                            if hasattr(grad_norm, "item"):
```

Comment:
```suggestion
                            if torch.is_tensor(grad_norm):
```

my bad, this just seems neater on second look